File: SOC/src/cc/game/soc/swing2/PopupButton.java
package cc.game.soc.swing2;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import cc.lib.swing.EZButton;

/**
 * Created by chriscaron on 2/28/18.
 */

public class PopupButton extends EZButton implements ActionListener {

    PopupButton(String txt) {
        super(txt, null);
        this.addActionListener(this);
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        doAction();
    }

    boolean doAction() {
        return true;
    }
}

File: SwingApps/apps/cc/applets/soc/PopupButton.kt
package cc.applets.soc

import cc.lib.swing.AWTButton
import java.awt.event.ActionEvent
import java.awt.event.ActionListener

/**
 * Created by chriscaron on 2/28/18.
 */
open class PopupButton internal constructor(txt: String) : AWTButton(txt, null), ActionListener {
	override fun actionPerformed(e: ActionEvent) {
		doAction()
	}

	open fun doAction(): Boolean {
		return true
	}
}